saleor/graphql/product/bulk_mutations/product_variant_bulk_update.py

```diff
@@ -0,0 +1,322 @@
+from collections import defaultdict
+from typing import cast
+
+import graphene
+from django.core.exceptions import ValidationError
+from django.db import transaction
+from graphene.utils.str_converters import to_camel_case
+
+from ....core.permissions import ProductPermissions
+from ....core.tracing import traced_atomic_transaction
+from ....product import models
+from ....product.error_codes import ProductErrorCode
+from ....product.search import update_product_search_vector
+from ....product.tasks import update_product_discounted_price_task
+from ....warehouse import models as warehouse_models
+from ...channel import ChannelContext
+from ...core.enums import ErrorPolicyEnum
+from ...core.types import BulkProductError, NonNullList, ProductVariantBulkError
+from ...core.utils import get_duplicated_values
+from ...plugins.dataloaders import get_plugin_manager_promise
+from ..utils import get_used_variants_attribute_values
+from .product_variant_bulk_create import (
+    BulkAttributeValueInput,
+    ProductVariantBulkCreate,
+    ProductVariantBulkCreateInput,
+    ProductVariantBulkResult,
+)
+
+
+class ProductVariantBulkUpdateInput(ProductVariantBulkCreateInput):
+    id = graphene.ID(description="ID of the product variant to update.", required=True)
+    attributes = NonNullList(
+        BulkAttributeValueInput,
+        required=False,
+        description="List of attributes specific to this variant.",
+    )
+
+
+class ProductVariantBulkUpdate(ProductVariantBulkCreate):
+    count = graphene.Int(
+        required=True,
+        default_value=0,
+        description="Returns how many objects were updated.",
+    )
+    results = NonNullList(
+        ProductVariantBulkResult,
+        required=True,
+        default_value=[],
+        description="List of the updated variants.",
+    )
+
+    class Arguments:
+        error_policy = ErrorPolicyEnum(
+            required=False,
+            default_value=ErrorPolicyEnum.REJECT_EVERYTHING.name,
+            description="Policies of error handling.",
+        )
+        variants = NonNullList(
+            ProductVariantBulkUpdateInput,
+            required=True,
+            description="Input list of product variants to update.",
+        )
+        product_id = graphene.ID(
+            description="ID of the product to update the variants for.",
+            name="product",
+            required=True,
+        )
+        error_policy = ErrorPolicyEnum(
+            required=False,
+            default_value=ErrorPolicyEnum.REJECT_EVERYTHING.value,
+            description="Policies of error handling.",
+        )
+
+    class Meta:
+        description = "Update a given product variants."
+        permissions = (ProductPermissions.MANAGE_PRODUCTS,)
+        error_type_class = BulkProductError
+        error_type_field = "bulk_product_errors"
```

Comment:
We could skip deprecated errors in new mutations. 
```suggestion
        error_type_class = BulkProductError
```